### PR TITLE
Uses progress tracker

### PR DESCRIPTION
**Pull request by cuducos**
Opening it as a draft as I would like to review it and think about what tests are needed (any ideas?).

### CODE REVIEW

File: progress.go [hunk 1]
```diff
@@ -138,6 +138,17 @@ func (p *progress) close() error {
 	return nil // Either not empty or error, suits both cases
 }
 
+// calculates the number of bytes downloaded
+func (p *progress) downloadedBytes() int64 {
+	var downloaded int64
+	for idx := range p.Chunks {
+		if p.Chunks[idx] == 1 {
+			downloaded += p.ChunkSize
```
Comment (cuducos) [inline progress.go:146]: This is not precise in the case of the last chunk. Let's say a file is 10 bytes long, and the chunk size is 4. The third chunk would be 2 bytes long, not 4.

Comment (danielfireman) [reply progress.go:146]: That means we need to track the number of bytes downloaded too. Right?

Comment (danielfireman) [reply progress.go:146]: My suggestion is to write the number of bytes downloaded per chunk, instead of simply 0 or 1. What do you think?

Comment (cuducos) [reply progress.go:146]: Good idea!

Comment (danielfireman) [reply progress.go:146]: On it.

**Review by danielfireman — APPROVED** [review]

**Pull request merged**